src/components/FromBob/bobConversionUtils.ts

```diff
@@ -0,0 +1,242 @@
+// File to hold functions which aid the conversion of bob files
+//  into our widget format
+
+import log from "loglevel";
+import convert from "xml-js";
+
+import { WidgetDescription } from "../Positioning/positioning";
+import { WidgetActions, WRITE_PV } from "../../widgetActions";
+
+interface BobDescription {
+  [key: string]: any;
+}
```

Comment:
Can we make this definition more useful? We know that the BobDescription will look like the compact JSON produced by the xml2js function, so could this be used to make it clearer what we expect to see?

Reply:
Have expanded this out a bit, could go on forever but perhaps not sensible to try and capture the entire bob schema within this interface?